spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/lock/JdbcClient.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.jdbc.lock;
+
+import java.io.Closeable;
+import java.util.Date;
+import java.util.UUID;
+
+import javax.sql.DataSource;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DuplicateKeyException;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Isolation;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.util.Assert;
+
+/**
+ * Encapsulation of the SQL shunting that is needed for locks. A {@link JdbcLockRegistry}
+ * needs a reference to a spring-managed (transactional) client service, so this component
+ * has to be declared as a bean.
+ *
+ * @author Dave Syer
+ *
+ */
+@Component
+@Transactional
+public class JdbcClient implements Closeable {
+
+	/**
+	 * Default value for the table prefix property.
+	 */
+	public static final String DEFAULT_TABLE_PREFIX = "INT_";
+
+	private enum Query {
+
+		DELETE("DELETE FROM %SLOCK WHERE REGION=? AND LOCK_KEY=? AND CLIENT_ID=?"),
+
+		DELETE_EXPIRED("DELETE FROM %SLOCK WHERE REGION=? AND LOCK_KEY=? AND CLIENT_ID=? AND CREATED_DATE<?"),
+
+		DELETE_ALL("DELETE FROM %SLOCK WHERE REGION=? AND CLIENT_ID=?"),
+
+		UPDATE("UPDATE %SLOCK SET CREATED_DATE=? WHERE REGION=? AND LOCK_KEY=? AND CLIENT_ID=?"),
+
+		INSERT("INSERT INTO %SLOCK (REGION, LOCK_KEY, CLIENT_ID, CREATED_DATE) VALUES (?, ?, ?, ?)"),
+
+		COUNT("SELECT COUNT(REGION) FROM %SLOCK WHERE CLIENT_ID=? AND CREATED_DATE>=?");
```

Comment:
OK. I think `REGION` has permission to live since the PK constraint is like:
```
constraint LOCK_PK primary key (LOCK_KEY, REGION)
```
So, I think this `SELECT` misses `REGION` restriction.